ios/plistcodec.go

```diff
@@ -56,3 +57,50 @@ func (plistCodec PlistCodec) Decode(r io.Reader) ([]byte, error) {
 	}
 	return payloadBytes, nil
 }
+
+type PlistCodecReadWriter struct {
+	w io.Writer
+	r io.Reader
+}
+
+func NewPlistCodecReadWriter(r io.Reader, w io.Writer) PlistCodecReadWriter {
+	return PlistCodecReadWriter{
+		w: w,
+		r: r,
+	}
+}
+
+func (p PlistCodecReadWriter) Write(m interface{}) error {
+	stringContent := ToPlist(m)
+	log.Tracef("Lockdown send %v", reflect.TypeOf(m))
+	buf := new(bytes.Buffer)
+	length := len(stringContent)
+	messageLength := uint32(length)
+
+	err := binary.Write(buf, binary.BigEndian, messageLength)
+	if err != nil {
+		return err
```

Comment:
please wrap the error with the struct and func name, like
PlistCodecReadWriter.Write: